designer/src/state/migrateNotebook.ts

```diff
@@ -65,9 +68,23 @@ export class ValidationError extends Error {
   }
 }
 
+/**
+ * Remove any fields that are explicitly set to null from `ui-specification.fields`.
+ * These fields will otherwise cause validation failures if the schema does not allow null.
+ */
+function removeNullFields(notebook: Notebook) {
+  if (!notebook['ui-specification']?.fields) return;
+
+  for (const fieldName of Object.keys(notebook['ui-specification'].fields)) {
+    if (notebook['ui-specification'].fields[fieldName] === null) {
+      delete notebook['ui-specification'].fields[fieldName];
+    }
+  }
+}
+
```

Comment:
could you also migrate the issue causing this bug which was the condition property being null, not the field itself?